### DISCUSSION TITLE
Selection of non-rectangular region for analysis in EBSD

### DISCUSSION THREAD

**Discussion by rrkamath** [mtex-toolbox/mtex#1193] [Ask Anything] [answered]
Hi all,

I was able to use the "Region" command (along with the "inpolygon" condition) to select a rectangular region for analysis from my EBSD dataset. I wanted to define a semi-elliptical/triangular region (cross-section of weld pools) for my analysis and couldn't find a way listed to do so in the documentation. Could anyone please let me know the command / code used to accomplish this ?

Thanks !
Rakesh

**Comment by nyyssont** [comment, marked as answer]
You want selectpolygon. Try this:

```
mtexdata martensite

plot(ebsd)
poly = selectPolygon;
ind = inpolygon(ebsd, poly);
close all

%check:
plot(ebsd(ind))

%data for your analysis:
ebsd_analysis = ebsd(ind);
```

**Reply by rrkamath** [reply]
Thank you ! That was exactly the thing I was looking for.

Rakesh
